tests/sdk/nodejs/preview-apply/index.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as k8s from "@pulumi/kubernetes";
+
+// Create provider with SSA enabled.
+var k8sProvider = new k8s.Provider("k8s", { enableServerSideApply: true });
+
+const dep = new k8s.apps.v1.Deployment(
+    "nginx-dep",
+    {
+        metadata: {
+            labels: {
+                app: "nginx",
+            },
+        },
+        spec: {
+            replicas: 1,
+            selector: {
+                matchLabels: {
+                    app: "nginx",
+                },
+            },
+            template: {
+                metadata: {
+                    labels: {
+                        app: "nginx",
+                    },
+                },
+                spec: {
+                    containers: [
+                        {
+                            name: "nginx",
+                            image: "nginx:latest",
+                            ports: [
+                                {
+                                    containerPort: 80,
+                                },
+                            ],
+                        },
+                    ],
+                },
+            },
+        },
+    },
+    { provider: k8sProvider }
+);
+
+const svc = new k8s.core.v1.Service(
+    "nginx-svc",
+    {
+        metadata: {
+            labels: {
+                app: "nginx",
+            },
+        },
+        spec: {
+            type: "LoadBalancer",
+            ports: [
+                {
+                    port: 80,
+                    targetPort: 80,
+                }
+            ],
+            selector: {
+                app: "nginx",
+            },
+        },
+    },
+    { provider: k8sProvider }
+);
+
+const ip = svc.status.apply((s) => s.loadBalancer.ingress[0].ip);
```

Comment:
To make the intent clearer, can we just assert that this has not run during preview?

Reply:
Do you have an example of how to do this?

Reply:
I was thinking something like this:

I don't remember the syntax to raise a TS exception, so this might be wrong.
```suggestion
if (pulumi.runtime.isDryRun() && !pulumi.isUnknown(svc.status)) {
    throw new Exception("Failed");
}
```

Reply:
I'm more inclined to having this intent in our test code, rather than within the Pulumi program. This way, the Pulumi program is representative to what our users normally specify within their code.